[qob] Batch backend fails to deserialize IR larger than 20MB

### What happened?

IR can get quite big, especially as it can contain an arbitrary amount of encoded literals from the user's python session. This was a (controversial) restriction imposed by Jackson and should be lifted.

### Version

0.2.131

### Relevant log output

```shell
Caused by: com.fasterxml.jackson.core.exc.StreamConstraintsException: String length (20026295) exceeds the maximum length (20000000)
```

--- a/hail/build.sc
+++ b/hail/build.sc
@@ -219,6 +219,7 @@ object main extends RootModule with HailScalaModule { outer =>
     Deps.Spark.core().excludeOrg("org.scalanlp"),  // Hail has an explicit dependency on Breeze 1.1
     Deps.Spark.mllib().excludeOrg("org.scalanlp"),  // Hail has an explicit dependency on Breeze 1.1
     Deps.Breeze.core,
+    Deps.jackson,
   )
 
   override def assemblyRules: Seq[Rule] = super.assemblyRules ++ Seq(

--- a/hail/build.sc
+++ b/hail/build.sc
@@ -112,7 +112,7 @@ object Deps {
   val junixsocket = ivy"com.kohlschutter.junixsocket:junixsocket-core:2.6.1"
   val log4j = ivy"org.apache.logging.log4j:log4j-1.2-api:2.17.2"
   val hadoopClient = ivy"org.apache.hadoop:hadoop-client:3.3.4"
-  val jackson = ivy"com.fasterxml.jackson.core:jackson-core:2.14.2"
+  val jackson = ivy"com.fasterxml.jackson.core:jackson-core:2.15.2"
 
   object Plugins {
     val betterModadicFor = ivy"com.olegpy::better-monadic-for:0.3.1"

--- a/hail/src/main/scala/is/hail/backend/service/ServiceBackend.scala
+++ b/hail/src/main/scala/is/hail/backend/service/ServiceBackend.scala
@@ -36,6 +36,7 @@ import org.apache.log4j.Logger
 import org.json4s.{DefaultFormats, Formats}
 import org.json4s.JsonAST._
 import org.json4s.jackson.JsonMethods
+import com.fasterxml.jackson.core.StreamReadConstraints
 
 class ServiceBackendContext(
   val billingProject: String,
@@ -464,6 +465,10 @@ object ServiceBackendAPI {
     log.info("BatchConfig parsed.")
 
     implicit val formats: Formats = DefaultFormats
+
+    StreamReadConstraints.overrideDefaultStreamReadConstraints(
+      StreamReadConstraints.builder().maxStringLength(Integer.MAX_VALUE).build()
+    );
     val input = using(fs.openNoCompression(inputURL))(JsonMethods.parse(_))
     val rpcConfig = (input \ "config").extract[ServiceBackendRPCPayload]
 

--- a/hail/src/main/scala/is/hail/backend/service/ServiceBackend.scala
+++ b/hail/src/main/scala/is/hail/backend/service/ServiceBackend.scala
@@ -32,11 +32,11 @@ import java.io._
 import java.nio.charset.StandardCharsets
 import java.util.concurrent._
 
+import com.fasterxml.jackson.core.StreamReadConstraints
 import org.apache.log4j.Logger
 import org.json4s.{DefaultFormats, Formats}
 import org.json4s.JsonAST._
 import org.json4s.jackson.JsonMethods
-import com.fasterxml.jackson.core.StreamReadConstraints
 
 class ServiceBackendContext(
   val billingProject: String,

